### DISCUSSION TITLE
Detail Seite (zweite Seite) lässt sich nicht öffnen.

### DISCUSSION THREAD

**Discussion by Marvin678** [neuefische/questions#241] [Web] [answered]
Hallo Leute,
wenn ich auf meiner Seite auf show details klicke, die auf meine zweite Seite verweisen soll, kommt diese Fehlermeldung.

![Bildschirmfoto 2022-10-11 um 10 56 40](https://user-images.githubusercontent.com/108797569/195058696-2b143775-0752-45e6-8d5e-a17fab30cc1c.png)

Freue mich über jeden Tipp den ihr habt.

https://github.com/Marvin678/capstone-project/pull/6

**Comment by mbosselmann** [comment, marked as answer]
Hallo Marvin,
dein Problem hängt mit dem **query** zusammen, den du versuchst einzusetzen und der **URL** selbst, die ein falsches Zeichen beinhaltet. Du schickst bei deinem `fetch` keinen query mit, sondern selektierst den `query` (die `id` des recipes, die du in der URL für die Detailseite verwendest) aus deiner dynamischen Route heraus. Dafür kannst du mit `useRouter` arbeiten und musst den `fetch` entsprechend anpassen. Außerdem muss in der fetch URL vor dem api key ein `?` und kein `&` stehen, weil der api key der einzige parameter ist, den du zusätzlich mitschickst. Siehe meine Beispiel-Lösung, die du gerne ausprobieren kannst.

```js
import {useRouter} from 'next/router';
import {useEffect, useState} from 'react';

export default function Details() {
	const [data, setData] = useState(null);
	const router = useRouter();
	const {recipe} = router.query;

	useEffect(() => {
		async function showDetails() {
			const response = await fetch(
				`https://api.spoonacular.com/recipes/${recipe}/analyzedInstructions?apiKey=${process.env.NEXT_PUBLIC_SPOONACULAR_API_KEY}`
			);

			const json = await response.json();
			console.log(json);
			setData(json);
		}
		showDetails();
	}, [recipe]);

	return <h1>details</h1>;
}
```

Hat dir das schon weitergeholfen? :)

**Reply by Marvin678** [reply]
Nein leider nicht.

**Reply by mbosselmann** [reply]
Wir haben die Lösung nochmal zu zweit besprochen und sie funktioniert, wie von mir vorgeschlagen. :)
